src/operators_lazyproduct.jl

```diff
@@ -10,23 +10,34 @@ The factors of the product are stored in the `operators` field. Additionally a
 complex factor is stored in the `factor` field which allows for fast
 multiplication with numbers.
 """
-mutable struct LazyProduct{BL,BR,F,T} <: AbstractOperator{BL,BR}
+
+mutable struct LazyProduct{BL,BR,F,T,KTL,BTR} <: AbstractOperator{BL,BR}
     basis_l::BL
     basis_r::BR
     factor::F
     operators::T
-    function LazyProduct{BL,BR,F,T}(operators::T, factor::F=1) where {BL,BR,F,T}
+    ket_l::KTL
+    bra_r::BTR
+    function LazyProduct{BL,BR,F,T,KTL,BTR}(operators::T, factor::F=1) where {BL,BR,F,T,KTL,BTR}
         for i = 2:length(operators)
             check_multiplicable(operators[i-1], operators[i])
         end
-        new(operators[1].basis_l, operators[end].basis_r, factor, operators)
+        ket_l=[Ket(operator.basis_l) for operator in operators]
+        bra_r=[Bra(operator.basis_r) for operator in operators]
```

Comment:
Change to `Tuple`s seems unacceptable, for we need use them in `mul!`.

Reply:
I don't understand. The list of `ket_l` is a vector of kets. If there are different bases involved, accessing elements of it will be type-unstable. You can change that vector to a tuple, however, and things should still work fine. All you need is to be able to iterate `ket_l` in `mul!`.

Reply:
OK, I know your point. I'll try it.

Reply:
Sorry, I just find that I can edit the inside vector.

Reply:
Hi @david-pl , I have changed that. 
Thank you very much for your kind instructions.
Learned a lot.